libr/util/logging.c

```diff
@@ -0,0 +1,157 @@
+/* radare - LGPL - Copyright 2007-2013 - pancake */
+
+#define LOGGING_CONFIGSTR_SIZE 1024
+#define LOGGING_OUTPUTBUF_SIZE 8192
+
+#include <r_core.h>
+#include <stdarg.h>
+
+static RLoggingFuncdef cb_logfunc_print = NULL; // Function to call when outputting log string
+static int cfg_loglvl = R_LOGLVL_ERROR; // Logging level output
+static int cfg_logtraplvl = R_LOGLVL_FATALTRAP; // Logging trap level
+static bool cfg_logsrcinfo = false; // Print out debug source info with the output
+static bool cfg_logcolors = false; // Output colored log text based on level
+static char cfg_logfile[LOGGING_CONFIGSTR_SIZE] = ""; // Output text to filename
+static char output_buf[LOGGING_OUTPUTBUF_SIZE] = ""; // Big buffer for building the output string
+
+/* TODO: Might want to create a r_logging_use_envvars function so that programs
+   that use r_util without core/cons have some way of changing the above vars
+   to the environmental variables. Also create seperate setter functions */
+
+// Used in cb_logging_config_level to check if input is integers only
+static inline bool str_is_digits(char *str, int len) {
+	int i;
+	for (i = 0; i < len; i++) {
+		if (!isdigit (str[i]))
+			return false;
+	}
+	return true;
+}
+
+// cconfig.c configuration callback functions below
+int cb_logging_config_level(void *coreptr, void *nodeptr) {
+	RConfigNode *node = (RConfigNode*)nodeptr;
+	char *value = node->value;
+	int value_len = strnlen (value, LOGGING_CONFIGSTR_SIZE);
+	if (!str_is_digits (value, value_len))
```

Comment:
just use `strtol` instead of str_is_digits + atoi

Reply:
The cconfig callbacks need to return false if the passed value is invalid. If I use strtol I can't tell if the user passed 'asdf' or 0/LONG_MAX/LONG_MIN. Although re-reading the manpage I can use errno to figure it out with strtol. I'll take a peek.

Reply:
yes you can AFAIK. you can also use `endptr` argument to know if there are other chars after the parsed number (e.g. `1234abcd` endptr should end up on the `a` and you'd know the input wasn't completely numeric)